test/Microsoft.AspNetCore.SignalR.Redis.Tests/RedisDependencyInjectionExtensionsTests.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Text;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Xunit;
+
+namespace Microsoft.AspNetCore.SignalR.Redis.Tests
+{
+    public class RedisDependencyInjectionExtensionsTests
+    {
+        // No need to go too deep with these tests, or we're just testing StackExchange.Redis again :). It's the one doing the parsing.
+        [Theory]
+        [InlineData("testredis.example.com", "testredis.example.com", 0, null, false)]
+        [InlineData("testredis.example.com:6380,ssl=True", "testredis.example.com", 6380, null, true)]
+        [InlineData("testredis.example.com:6380,password=hunter2,ssl=True", "testredis.example.com", 6380, "hunter2", true)]
+        public void AddRedisWithConnectionStringProperlyParsesOptions(string connectionString, string host, int port, string password, bool useSsl)
+        {
+            var services = new ServiceCollection();
+            services.AddSignalR().AddRedis(connectionString);
```

Comment:
Do we need to worry about what happens if `AddRedis` is called multiple times? Ignore it? Throw error?

Reply:
Our general pattern is to allow that to happen. The way Options works is that each delegate that's registered to the container is run in the order of registration. So if I did:

```
.AddRedis(o => o.Options.EndPoints.Add("myredis.example.com"))
.AddRedis("adifferentredis.example.com")
.AddRedis(o => o.Options.Ssl = true);
```

The first line would run and add the endpoint, but the second line would overwrite the config (because we replace the `ConfigurationOptions` and StackExchange.Redis doesn't provide a way to merge them). The third line would **augment** the configuration to enable SSL.

You'd end up with `adifferentredis.example.com` and SSL enabled.

Reply:
ok!

Reply:
Calling these `.AddFoo` methods multiple times is also a common pattern when you want to override our default options (see how we call `.AddJsonProtocol` for you in [`.AddSignalR`](https://github.com/aspnet/SignalR/blob/dev/src/Microsoft.AspNetCore.SignalR/SignalRDependencyInjectionExtensions.cs#L12) but you can call it again to change your serializer settings).